### PR TITLE
[DOCS] `time_series_dimension` fields do not support `ignore_above`

### LINKED ISSUE #99044: Dimension fields and 'ignore_above' missing documentation
### Elasticsearch Version

8.7.0 and above

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

All

### Problem Description

Dimension fields in time series indices do not support `ignore_above`.
Mapping a field setting `time_series_dimension` and `ignore_above` results int the following exception

```
elasticsearch.BadRequestError: BadRequestError(400, 'mapper_parsing_exception', 'Failed to parse mapping: Field [ignore_above] cannot be set in conjunction with field [time_series_dimension]')
```

This behaviour is expected but not documented. We need to update documentation clarifying that using `ignore_above` is not allowed for dimension fields. Note also that with existing limitations on number and size of dimensions for time series indices and usage of synthetic source as the default, setting `ignore_above` is pretty much useless.

### Steps to Reproduce

Try to create a time series index whose mappings include a keyword field that uses `time_series_dimension` and `ignore_above`.

### Logs (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by lkts**
There is existing validation for this combination of parameters but it was not documented. I was wondering if we should explain _why_ is that but we don't really do that for other restrictions here.

Closes #99044

### CODE REVIEW

File: docs/reference/mapping/types/keyword.asciidoc [hunk 1]
```diff
@@ -165,6 +165,7 @@ Dimension fields have the following constraints:
 * The `doc_values` and `index` mapping parameters must be `true`.
 * Field values cannot be an <<array,array or multi-value>>.
 // end::dimension[]
+* <<ignore-above,`ignore_above`>> mapping parameter is not supported.
```
Comment (martijnvg) [inline docs/reference/mapping/types/keyword.asciidoc:168]: Maybe reword to: 
```
The <<ignore-above,`ignore_above`>> mapping attribute isn't supported.
```

**Review by martijnvg — COMMENTED** [review]
I think it is a good idea to explain why these listed options aren't supported.

The main reason is that dimension fields are used to create the tsid. A time series data stream also has synthetic source enabled. So the original document as was provided isn't kept around. If the a feature like ignore above or normalizer is enabled then what gets stored in doc values and inverted index is changed (either trimmed or altered). We will never be able to re-generate the tsid as the original data is no longer available.

After re-reading this paragraph, I realize with recent improvements, the following: `Field values cannot be larger than 1024 bytes.` is no longer true. So this can be removed. Additionally with recent improvements, we could decide to remove the following constraint: `Field values cannot be an array or multi-value`.

**Comment by lkts** [comment]
@martijnvg removed `1024 bytes` one and added short explanation

**Review by martijnvg — APPROVED** [review]
Thanks, LGTM!
